app/components/sidebars/main/channels_list/filtered_list/filtered_list.js

```diff
@@ -82,14 +86,13 @@ class FilteredList extends Component {
         return !deepEqual(this.props, nextProps, {strict: true}) || !deepEqual(this.state, nextState, {strict: true});
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.term !== nextProps.term) {
-            const {actions, currentTeam} = this.props;
-            const {term} = nextProps;
+    static getDerivedStateFromProps(props, state) {
+        if (state.term !== props.term) {
+            const {actions, currentTeam} = state;
+            const {term} = props;
             const {searchChannels, searchProfiles} = actions;
-            const dataSource = this.buildData(this.props, term);
+            const dataSource = this.buildData(state, term);
```

Comment:
`this` is going to be undefined when calling `this.buildData`, so that function needs to be reworked to be a static function, or this has to be done by the component some other way. Similarly, any other usage of `this` in `getDerivedStateFromProps` needs to be changed

Reply:
`this.builddata` internally uses state methods. so does all the method should be made static or the current instance can be made as a part of state like `currentState:this` and use it to call the builddata method